### ISSUE TITLE
Typescript can't find a module that has been mapped with SystemJS

### ISSUE THREAD

**Issue by maximedupre** [Microsoft/TypeScript#8924]
This *could* be related to #5039, I'm not entirely sure.

`analytics.ts`:

    import ga from 'googleAnalytics';

    ga('create', 'UA-77054312-1', 'auto');
    ga('send', 'pageview');

`googleAnalytics` isn't a node_modules and I've got `moduleResolution` set to `"node"`, so of course `tsc`  gives me this warning: `Cannot find module 'googleAnalytics'`

However, SystemJS knows where to find the module, because it has been mapped up front like so:

    map: {
        'googleAnalytics': '//www.google-analytics.com/analytics.js'
    }

There is a [type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/google.analytics) file for the google analytics library and I have it, but it doesn't fix my problem, I still get the error. Also, do I have to rename the `googleAnalytics` module/SystemJS path to `ga` in order to match the type definition file name (did that and it still doesn't work)?

**Comment by mhegazy** [comment]
the definition file does not state anywhere that googleAnalytics is a valid module. This is probably an issue with the definition file, but i am not familiar with the googleAnalytics  to tell for sure.

to be able to consume it as a module, the definition file should have something like:

```ts
declare module "googleAnalytics" {
    export = ga;
}
```

you can add this definition locally in one of your files, or add it to the one on definitllyTyped. I would say this is an issue for https://github.com/DefinitelyTyped/DefinitelyTyped.
